incubator/hnc/internal/validators/namespace.go

```diff
@@ -99,6 +106,16 @@ func (v *Namespace) handle(req *nsRequest) admission.Response {
 	return allow("")
 }
 
+func (v *Namespace) illegalExcludedNamespaceLabel(req *nsRequest) admission.Response {
+	for l := range req.ns.Labels {
+		if l == api.LabelExcludedNamespace && !config.ExcludedNamespaces[req.ns.Name] {
```

Comment:
What about if that namespace _already_ has the label? Should we block further changes to that namespace until the label is deleted?

Reply:
Yes, I updated the message to say "Please remove the %q label. See http://... for detail".

Reply:
I'm not a huge fan of this. I feel like webhooks should only validate the _changes_ in a resource, not what's already there. For example, imagine an automated process that's trying to make some unrelated change; we shouldn't block them. And we shouldn't have to ask the user to remove the label given that we've got a reconciler that can do it automatically.

Reply:
sg. Changed the message back to "You cannot exclude this namespace using the %q label. See..for detail"
and commented here with:
```
			// Note: this only blocks the request if it has a newly added illegal
			// excluded-namespace label because existing illegal excluded-namespace
			// label should have already been removed by our reconciler. For example,
			// even when the VWHConfiguration is removed, adding the label to a non-
			// excluded namespace would pass but the label is immediately removed; when
			// the VWHConfiguration is there but the reconcilers are down, any request
			// gets denied anyway.
```